webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/DisplayNode.scala

```diff
@@ -85,10 +86,21 @@ object DisplayNode extends Loggable {
   private[this] val uuidGen              = RudderConfig.stringUuidGenerator
   private[this] val nodeInfoService      = RudderConfig.nodeInfoService
   private[this] val linkUtil             = RudderConfig.linkUtil
+  private[this] val asyncComplianceService    = RudderConfig.asyncComplianceService
   private[this] val deleteNodePopupHtmlId = "deleteNodePopupHtmlId"
   private[this] val errorPopupHtmlId = "errorPopupHtmlId"
   private[this] val successPopupHtmlId = "successPopupHtmlId"
 
+
+  private def loadComplianceBar(nodeInfo: Option[NodeInfo]): Option[JsArray] = {
+    for {
+      node       <- nodeInfo
+      t          <- asyncComplianceService.nodeCompliance(node.id)
+    } yield {
+      ComplianceLevelSerialisation.ComplianceLevelToJs(t(node.id).get).toJsArray()
```

Comment:
never use get !! this thows exception ! in fact your service should return a Box[ComplianceLevel], not the Map[NodeId,Option[ComplianceLevel]]

Also don't do t(node.id), it's also a get and it throws exception !!